### ISSUE TITLE
Advanced Search (https://bl.vuexistapps.us/search.html)

### ISSUE THREAD

**Issue by davidamichelson** [srophe/britishLibrary#117]
Please add the following functions and boxes to advanced search. (Note, currently the search does not return anything).

- [ ] Keyword: this box should search by string in all textnodes EXCEPT those that are contained in TEI:encodingDesc and TEI:profileDesc and children of those elements.
- [ ] Author: this box should search by string in all textnodes of all TEI:Author EXCEPT those that are contained in TEI:additional and children of that element.  (How do we want to hand URI search for Author? Could we allow this box to take strings or URIS and handle them differently?) (@wlpotter I didn't limit this to msItem because it could be that item in additions also has an author?)
- [ ] Translated Title: this box should search by string in all textnodes of TEI:title and TEI:finalRubric EXCEPT those that are contained in TEI:additional, TEI:encodingDesc, and TEI:profileDesc and children of those elements. (How do we want to hand URI search for works? Could we allow this box to take strings or URIS and handle them differently?) (@wlpotter do some TEI:finalRubrics have English? I think so, right?)
- [ ] Syriac Title: this box should search by string in all textnodes of TEI:rubric and TEI:finalRubric (EXCEPT those that are contained in TEI:additional, TEI:encodingDesc, and TEI:profileDesc and children of those elements, which should not exist). (How do we want to hand URI search for works? Could we allow this box to take strings or URIS and handle them differently?)
- [ ] Locus: this box should search by numeric string only for exact match only in the //locus/@to and //locus/@from attributes
- [ ] Date: Let's discuss if perhaps the results could have a slider instead of a box for this.
- [ ] Script: Let's discuss if perhaps the results can be faceted by this instead of a box.
- [ ] Material: Let's discuss if perhaps the results can be faceted by this instead of a box.

**Comment by wlpotter** [comment]
@davidamichelson here's a bit more context on authors in additions and finalRubrics that have English-language content

- Authors in additions: we have one record where this is the case (https://bl.vuexistapps.us/manuscript/26, in addition 2), and I'm not sure the data is correct here. I think anything that would have an author in the sense used in this advanced search would (or should) be included in msItem rather than additions?
- final rubrics with English-language content: we have two cases, I think. In https://bl.vuexistapps.us/manuscript/10 we have the translated final rubric for item 1; in https://bl.vuexistapps.us/manuscript/25 we have the translated final rubric. There are also a few cases of tei:foreign or tei:note elements that are `@xml:lang="en"`. I think we do want to include this in the search for translated titles

----

I also wonder if, for the URI search for works or authors we could do something similar to [TSG's advanced search](https://syriaca.org/geo/search.html) where you can narrow results by including a related person or work URI? The trick would be scoping this, I suppose, so the URI appears in the same msItem or additions context as the search string(s)? This might need further discussion

**Comment by wsalesky** [comment]
@davidamichelson @wlpotter For the Locus search, do you want a box for to and a box for from? So you can search a range?

**Comment by wsalesky** [comment]
@wsalesky Note to self, it may be easier to search by title/author fields that are constructed at index time.

**Comment by davidamichelson** [comment]
> @davidamichelson @wlpotter For the Locus search, do you want a box for to and a box for from? So you can search a range?

Sure! Let's have three boxes From, To, and Exact Folio

Note, for these, please only run the search for the numeric value of the search term (we will ignore "a" "b" "r" "v" in 21b or 34v)

**Comment by davidamichelson** [comment]
On second thought, don't create a locus search for now!

**Comment by wsalesky** [comment]
@davidamichelson @wlpotter We forgot to discuss:
Date: Let's discuss if perhaps the results could have a slider instead of a box for this.
 Script: Let's discuss if perhaps the results can be faceted by this instead of a box.
 Material: Let's discuss if perhaps the results can be faceted by this instead of a box.

If you give me the xpath I can go ahead and make facets.

**Comment by wlpotter** [comment]
Here are those xpaths:

- Date: `/TEI/teiHeader/fileDesc/sourceDesc/msDesc/history/origin/origDate`. We will want to use the `@notBefore`, `@notAfter`, `@from`, `@to`, and  `@when` attributes.
  - Note that there is sometimes an `//origDate` element with `@notBefore-custom`, etc. attributes but we don't want to use these for the slider
- Script: `/TEI/teiHeader/fileDesc/sourceDesc/msDesc/physDesc/handDesc/handNote/@script`. We would want to use the labels from [this comment](https://github.com/srophe/britishLibrary/issues/62#issue-1658870275).
- Material: `/TEI/teiHeader/fileDesc/sourceDesc/msDesc/physDesc/objectDesc/supportDesc/@material`. We would want the labels from [this comment](https://github.com/srophe/britishLibrary/issues/57#issue-1658832846) under "Material Support"

Let me know if you need anything further here

**Comment by davidamichelson** [comment]
@wsalesky please proceed with Date, Script, and Material as facets and let's discuss if we also add them to search after that.

**Comment by wsalesky** [comment]
@davidamichelson @wlpotter Can I have some test cases to use for the 'limit by' part of the search? at least one test case per limit would be great!

**Comment by wsalesky** [comment]
@davidamichelson @wlpotter I have added everything but the 'Limit by' function. Please test and send any bugs. Thanks!

**Comment by wsalesky** [comment]
@davidamichelson @wlpotter for the 'Search in ' function, do you really want to limit all of the above options by these paths? Or just the keyword? For example, just search persons and places in /TEI/teiHeader/fileDesc/sourceDesc/msDesc/msPart/history  ?

**Comment by davidamichelson** [comment]
> @davidamichelson @wlpotter Can I have some test cases to use for the 'limit by' part of the search? at least one test case per limit would be great!

We have split this off as its own issue and area waiting on this. No action needed.
